### ISSUE TITLE
@testing-library/react's `fireEvent` no longer triggers useGesture callbacks as of 8.0.0

### ISSUE THREAD

**Issue by stevenpetryk** [pmndrs/react-use-gesture#224]
As of 8.0.0, `fireEvent...` calls seem to no longer trigger callbacks inside of react-use-gesture hooks. I've tried with a few different hooks—`useDrag`, `useHover`, and `useGesture`, and I can't get the callbacks to fire in tests at all. Works fine in v7 :)

Here's a [minimal reproduction repository](https://github.com/stevenpetryk/react-use-gesture-test-failure-repro). 

Quick links: 

- [README](https://github.com/stevenpetryk/react-use-gesture-test-failure-repro#readme)
- [Component with hook](https://github.com/stevenpetryk/react-use-gesture-test-failure-repro/blob/master/src/App.js) 
- [Failing test](https://github.com/stevenpetryk/react-use-gesture-test-failure-repro/blob/master/src/App.test.js)

**Comment by dbismut** [comment]
@stevenpetryk as per the breaking changes v8 now uses PointerEvents wherever it can. Have a look at how we cope with jsdom not being able to fire PointerEvents properly.

**Comment by stevenpetryk** [comment]
I totally get the assumption that I didn't read the breaking changes, as a lot of garbage issues are opened on popular repos—but I did read them, and I read through the docs, but still wasn't able to figure this out. One of the first things I did was try to emulate this repo's tests, and spent some time confused about why `fireEvent.mosueEnter` works in `Hover.test.tsx`, for example, but not in my repo.

What I'm saying is, I'm someone who did my homework before opening the issue and _still_ couldn't crack the code, so might it be worth adding to the FAQ? :)

**Comment by dbismut** [comment]
No worries! I could spot this wasn’t a garbage issue and that’s there was indeed some background checks before posting, sorry if that came across differently!

I'm not sure this is the best solution, but here is how we cope with PointerEvents not being handled properly by jsdom. Probably some code I've stolen from elsewhere, looks too clever:

1. Patch `addEventListener` so `pointerxxx` events are transformed into `mousexxx` events
https://github.com/pmndrs/react-use-gesture/blob/master/setupPointerEvent.js

2. Patching `createEvent` from `@testing-library` at the beginning of each test file (not sure why I didn't factor this into one file)
https://github.com/pmndrs/react-use-gesture/blob/a7d24e382c1926f87a6ce7bcf6dbc1401e4e066f/test/Api.test.tsx#L12-L25

Maybe I could try exporting a setup test file that would patch `@testing-library` as part of this repo? Not sure how to do this, maybe you have an idea?

**Comment by stevenpetryk** [comment]
Strangely, I was able to get it to work with just step 1, but I removed `document.createEvent = null`. It created a "state change on unmounted component" warning, but the tests still seemed to pass just fine. Using both 1 and 2 caused most tests to error, saying `document.createEvent is not a function`.

I'll check out some other popular Jest companions and see how they patch things.

**Comment by stevenpetryk** [comment]
Interestingly, when I added this code to a large project, it caused a memory leak. Seems to involve the fact that `addEventListener` is getting patched.

**Comment by dbismut** [comment]
For sure let me know how it goes. I'd be more than happy to fix this.

**Comment by adambulmer** [comment]
i am also seeing this issue as part of v9. I've tried the steps above and get the following error. ```Cannot destructure property `pointerId` of 'undefined' or 'null'.```

**Comment by dbismut** [comment]
@adambulmer sorry can you rephrase what your problem is and what are the "steps above"?

**Comment by dbismut** [comment]
@AdrienLemaire sorry, this is too abstract for me, I'd need more details about the failing tests, error messages etc. All I can tell is that tests work fine on the repo.

**Comment by marcusstenbeck** [comment]
I finally sorted this after a few hours after discovering that my `config.transform()` ended up returning invalid values. 

[This line](https://github.com/pmndrs/use-gesture/blob/main/packages/core/src/engines/Engine.ts#L230) will cause `state.intentional` to be `false`, and then fail silently.

**Comment by dbismut** [comment]
Feel free to submit a PR, I'm open to it.

**Comment by marcusstenbeck** [comment]
@dbismut Done @ #382

**Comment by dbismut** [comment]
@marcusstenbeck released in `@10.1.3`, thanks for your help.
